artifactory/services/utils/searchutil.go

```diff
@@ -74,13 +87,23 @@ func SearchBySpecWithAql(specFile *ArtifactoryCommonParams, flags CommonConf, re
 		return nil, err
 	}
 
-	// Filter results by build.
-	if specFile.Build != "" && len(results) > 0 {
-		// If requiredArtifactProps is not NONE and 'includePropertiesInAqlForSpec' for specFile returned true, results contains properties for artifacts.
-		resultsArtifactsIncludeProperties := requiredArtifactProps != NONE && includePropertiesInAqlForSpec(specFile)
-		results, err = filterAqlSearchResultsByBuild(specFile, results, flags, resultsArtifactsIncludeProperties)
-		if err != nil {
-			return nil, err
+	if len(results) > 0 {
+		// Filter results by build.
+		if specFile.Build != "" {
+			// If requiredArtifactProps is not NONE and 'includePropertiesInAqlForSpec' for specFile returned true, results contains properties for artifacts.
+			resultsArtifactsIncludeProperties := requiredArtifactProps != NONE && includePropertiesInAqlForSpec(specFile)
+			results, err = filterAqlSearchResultsByBuild(specFile, results, flags, resultsArtifactsIncludeProperties)
+			if err != nil {
+				return nil, err
+			}
+		}
+
+		// Filter results by bundle.
+		if specFile.Bundle != "" {
+			results, err = filterAqlSearchResultsByBundle(specFile, results, flags)
```

Comment:
It looks like we're using a two AQLs approach when the spec includes a bundle. Do we have to do this? I know that we had to tale this approach for "builds" due to performance reasons, but for bundles, can we have one AQL to include the bundles?

Reply:
This is a difficult question. Maybe @galusben will know the answer?